app/models/form_submission_attempt.rb

```diff
@@ -25,15 +25,31 @@ class FormSubmissionAttempt < ApplicationRecord
 
     event :fail do
       after do
+        user_uuid = form_submission.user_account.id
+        form_type = form_submission&.form_type
+        log_info = { form_submission_id:, benefits_intake_uuid: form_submission&.benefits_intake_uuid, form_type:,
+                     user_uuid: user_uuid }
+        zsf_function = "#{form_id} form submission to Lighthouse"
         if should_send_simple_forms_email
-          Rails.logger.info({
-                              message: 'Preparing to send Form Submission Attempt error email',
-                              form_submission_id:,
-                              benefits_intake_uuid: form_submission.benefits_intake_uuid,
-                              form_type: form_submission.form_type
-                            })
+          Rails.logger.info('Preparing to send Form Submission Attempt error email', log_info)
           simple_forms_enqueue_result_email(:error)
+        elsif should_send_form526_form4142_email
+          zsf_function = 'Form 526 Flow - Form 4142 failure email queuing'
+          form526_submission_id = Form526Submission.find_by(saved_claim_id:).id
+          Rails.logger.info('Queuing Form526:Form4142 failure email to VaNotify',
+                            log_info.merge({ form526_submission_id: }))
+          jid = EVSS::DisabilityCompensationForm::Form4142DocumentUploadFailureEmail.perform_async form526_submission_id
+          Rails.logger.info('Queuing Form526:Form4142 failure email to VaNotify completed', log_info.merge({ jid: }))
         end
+      rescue => e
+        cl = caller_locations.first
+        call_location = ZeroSilentFailures::Monitor::CallLocation.new(zsf_function, cl.path, cl.lineno)
```

Comment:
The `zsf_function` string is specific to 'Form 526 Flow - Form 4142', but might also rescue failures related to Simple Forms, correct?

Reply:
`zsf_function` is defined on line 32, 

```
        zsf_function = "#{form_type} form submission to Lighthouse"
```

and then conditionally inside this if statement here. So, if going this path, use my newly set one, if going the other path, use the initially set one, which is the variable they were previously using

Reply:
> but might also rescue failures related to Simple Forms, correct?

Correct, which is why it is defined variably. 

If it rescues from the simple forms path, it is already initially set to their value. 

If it goes into our code (the elsif condition) changing that value is the very first thing in that elsif block so if it errors/rescues from our way, it will log our thing.

Reply:
I missed that this was a reassignment.